### ISSUE TITLE
Hilt viewmodel + replace + transition = crash

### ISSUE THREAD

**Issue by Tolriq** [adrielcafe/voyager#17]
So this is a tricky one.

```kotlin
Navigator(Screen1()) { navigator ->
                        SlideTransition(navigator)
                    }

class Screen1() : AndroidScreen {
    @Composable
    override fun Content() {
            val viewModel = getViewModel<Screen1ViewModel>()
            ....
            LocalNavigator.currentOrThrow.replace(Screen2())
    }
}

``` 

Will lead to a crash
```
    java.lang.IllegalArgumentException: SavedStateProvider with the given key is already registered
        at androidx.savedstate.SavedStateRegistry.registerSavedStateProvider(SavedStateRegistry.java:111)
        at androidx.lifecycle.SavedStateHandleController.attachToLifecycle(SavedStateHandleController.java:50)
        at androidx.lifecycle.SavedStateHandleController.create(SavedStateHandleController.java:70)
        at androidx.lifecycle.AbstractSavedStateViewModelFactory.create(AbstractSavedStateViewModelFactory.java:67)
        at androidx.lifecycle.AbstractSavedStateViewModelFactory.create(AbstractSavedStateViewModelFactory.java:84)
        at dagger.hilt.android.internal.lifecycle.HiltViewModelFactory.create(HiltViewModelFactory.java:109)
```

From a quick look the viewmodel is properly cleared, but it seems Voyager tries to re access the SaveStateProvider later with the same key (obviously as it's the same composable).

There's no issue without enabling transitions, so this looks like some wrong order of calls, whatever access the savestate during the transition should do it before the full removal of the composable to avoid trying to reuse the key.
There's also no problem when just pushing a new screen. This works without using hitviewmodel.

**Comment by programadorthi** [comment]
Fixed just putting a missed remember on ViewModel creation here #18. [Checkout solution](https://github.com/adrielcafe/voyager/pull/18/commits/e9bdc00b3a2fdf8f91296a27a825506ed73720f8)

**Comment by adrielcafe** [comment]
Fixed on [1.0.0-beta13](https://github.com/adrielcafe/voyager/releases/tag/1.0.0-beta13), feel free to reopen if the issue persists.
